src/V3HierBlock.cpp

```diff
@@ -0,0 +1,390 @@
+    for (HierBlockSet::const_iterator child = m_children.begin(); child != m_children.end();
+         ++child) {
+        const bool deleted = (*child)->m_children.erase(this);
+        UASSERT_OBJ(deleted, m_modp, " is not registered");
+    }
+}
+
+V3StringList V3HierBlock::commandOptions(bool forCMake) const {
+    V3StringList opts;
+    const string prefix = "V" + modp()->name();
+    opts.push_back(" --prefix " + prefix);
+    opts.push_back(" --mod-prefix " + prefix);
+    opts.push_back(" --top-module " + modp()->name());
+    opts.push_back(" --protect-lib " + modp()->name());  // mangled name
+    opts.push_back(" --protect-key " + v3Global.opt.protectKeyDefaulted());
+    opts.push_back(" --hierarchical-mode");
+
+    const StrGParams gparamsStr = stringifyParams(gparams(), forCMake, true);
+    for (StrGParams::const_iterator paramIt = gparamsStr.begin(); paramIt != gparamsStr.end();
+         ++paramIt) {
+        opts.push_back("-G" + paramIt->first + "=" + paramIt->second + "");
+        if (forCMake) opts.back() = escapeStringForCMake(opts.back());
+    }
+    return opts;
+}
+
+V3StringList V3HierBlock::hierBlockOptions(bool forCMake) const {
+    V3StringList opts;
+    const StrGParams gparamsStr = stringifyParams(gparams(), forCMake, false);
+    opts.push_back("--hierarchy-block ");
+    string s = modp()->origName();  // origName
+    s += "," + modp()->name();  // mangledName
+    for (StrGParams::const_iterator paramIt = gparamsStr.begin(); paramIt != gparamsStr.end();
+         ++paramIt) {
+        s += "," + paramIt->first;
+        s += "," + paramIt->second;
+    }
+    if (forCMake) s = escapeStringForCMake(s);
+    opts.back() += s;
+    return opts;
+}
+
+string V3HierBlock::hierPrefix() const { return "V" + modp()->name(); }
+
+string V3HierBlock::hierSomeFile(bool withDir, const char* prefix, const char* suffix) const {
+    string s;
+    if (withDir) s = hierPrefix() + '/';
+    s += prefix + modp()->name() + suffix;
+    return s;
+}
+
+string V3HierBlock::hierWrapper(bool withDir) const { return hierSomeFile(withDir, "", ".sv"); }
+
+string V3HierBlock::hierMk(bool withDir) const { return hierSomeFile(withDir, "V", ".mk"); }
+
+string V3HierBlock::hierLib(bool withDir) const { return hierSomeFile(withDir, "lib", ".a"); }
+
+string V3HierBlock::hierGenerated(bool withDir) const {
+    return hierWrapper(withDir) + ' ' + hierMk(withDir);
+}
+
+//######################################################################
+class V3HierBlockPlan::Visitor : public AstNVisitor {
+    typedef std::set<const AstModule*> ModuleSet;
+    V3HierBlockPlan* const m_planp;
+    AstModule* m_modp;
+    ModuleSet m_referred;  // Modules that have hier_block pragma
+    ModuleSet m_checked;  // Modules that are already checked;
+    V3HierBlock::GParams m_gparams;  // list of variables that is AstVarType::GPARAM
+
+    virtual void visit(AstNode* nodep) VL_OVERRIDE { iterateChildren(nodep); }
```

Comment:
```suggestion
    virtual void visit(AstNodeMath*) VL_OVERRIDE {}  // Accelerate
    virtual void visit(AstNode* nodep) VL_OVERRIDE { iterateChildren(nodep); }
```

And move under all other visitors (to match general style).